docs/setup-guide/build_k8s_from_source/setup_1.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+echo "Setup Part 1 started... \n"
+
+echo "Configuring iptables..."
+sudo modprobe br_netfilter
+cat <<EOF | sudo tee /etc/modules-load.d/k8s.conf
+br_netfilter
+EOF
+cat <<EOF | sudo tee /etc/sysctl.d/k8s.conf
+net.bridge.bridge-nf-call-ip6tables = 1
+net.bridge.bridge-nf-call-iptables = 1
+EOF
+sudo sysctl --system
+echo "[x] Finished configuring iptable.\n"
+
+echo "Installing runtime docker..."
+sudo apt-get -y update; sudo apt-get -y install docker.io
+sudo systemctl enable docker.service
+sudo groupadd docker
+sudo usermod -aG docker $USER
+
+echo "\nRebooting the machine..."
+echo "After reboot run Setup Part 2 script setup_2.sh"
+sleep 3
```

Comment:
is the sleep needed?

Reply:
oh i see. it's for user to see the reboot message.

Reply:
user might not be ready to do the reboot here yet (e.g they might have some vim session running in the background) so I'd suggest simply print out a message to ask for the user to reboot.

Reply:
Yes. The purpose of this 'sleep 3' is to let user have sufficient time to see the message.

Do you mean vim session for editing the config files we mentioned here, or just general vim sessions. The assumption here is the machine is a fresh installed instance and we automate the setup including modifying those configuration files (those 'tee' commands).

Reply:
general vim sessions. a sudden reboot even with a 3-second pause might be interruptive. with a new machine, the user might kick off the script and then switch to other place to come back later on. if it runs out of the 3-second window the user might come back finding the machine has already rebooted, which can be quite confusing.